networkx/algorithms/isomorphism/vf2pp.py

```diff
@@ -1,44 +1,84 @@
 import collections
 
 import networkx as nx
-from networkx.algorithms.isomorphism.VF2pp_helpers.candidates import find_candidates
-from networkx.algorithms.isomorphism.VF2pp_helpers.feasibility import feasibility
-from networkx.algorithms.isomorphism.VF2pp_helpers.node_ordering import matching_order
-from networkx.algorithms.isomorphism.VF2pp_helpers.state import (
-    restore_state,
-    update_state,
+from networkx.algorithms.isomorphism.vf2pp_helpers.candidates import _find_candidates
+from networkx.algorithms.isomorphism.vf2pp_helpers.feasibility import _feasibility
+from networkx.algorithms.isomorphism.vf2pp_helpers.node_ordering import _matching_order
+from networkx.algorithms.isomorphism.vf2pp_helpers.state import (
+    _restore_state,
+    _update_state,
 )
 
+__all__ = ["vf2pp_mapping", "vf2pp_is_isomorphic"]
 
-def VF2pp(G1, G2, G1_labels, G2_labels):
+
+def vf2pp_mapping(G1, G2, node_labels=None, default_label=None):
+    """Yields all the possible mappings between G1 and G2.
+
+    Parameters
+    ----------
+    G1,G2: NetworkX Graph or MultiGraph instances.
+        The two graphs to check for isomorphism or monomorphism.
+
+    node_labels: Label name
+        The label name of all nodes
+
+    default_label: Label name
+        Let the user pick a default label value
+    """
     try:
-        mapping = next(VF2pp_solver(G1, G2, G1_labels, G2_labels))
+        mapping = next(vf2pp_all_mappings(G1, G2, node_labels, default_label))
         return mapping
     except StopIteration:
         return None
 
 
-def VF2pp_solver(G1, G2, G1_labels, G2_labels):
+def vf2pp_is_isomorphic(G1, G2, node_labels=None, default_label=None):
+    """Examines whether G1 and G2 are isomorphic.
+
+    Parameters
+    ----------
+    G1,G2: NetworkX Graph or MultiGraph instances.
+        The two graphs to check for isomorphism or monomorphism.
+
+    node_labels: Label name
+        The label name of all nodes
+
+    default_label: Label name
+        Let the user pick a default label value
+
+    Returns
+    -------
+    True if the two graphs are isomorphic. False otherwise.
+    """
+    return True and not (not vf2pp_mapping(G1, G2, node_labels, default_label))
+
+
+def vf2pp_all_mappings(G1, G2, node_labels=None, default_label=None):
     """Implementation of the VF2++ algorithm.
 
     Parameters
     ----------
     G1,G2: NetworkX Graph or MultiGraph instances.
         The two graphs to check for isomorphism or monomorphism.
 
-    G1_labels,G2_labels: dict
-        The label of every node in G1 and G2 respectively.
+    node_labels: Label name
+        The label name of all nodes
+
+    default_label: Label name
+        Let the user pick a default label value
 
     Returns
     -------
     Node mapping, if the two graphs are isomorphic. None otherwise.
     """
+    G1_labels, G2_labels = dict(), dict()
```

Comment:
```suggestion
    G1_labels, G2_labels = {} ,{}
```

Reply:
Doesn't that create a set instead of a dict?

Reply:
Theoretically this should be a set but i tested it and works fine